src/connection_string.ts

```diff
@@ -569,7 +567,7 @@ export const OPTIONS = {
           if (['none', 'snappy', 'zlib'].includes(String(c))) {
             compressionList.add(String(c));
           } else {
-            throw new MongoParseError(`${c} is not a valid compression mechanism`);
+            throw new MongoCompressionError(`${c} is not a valid compression mechanism`);
```

Comment:
this particular compression error is an api error

Reply:
In hindsight, I think this may be better suited as InvalidArgument because the it seems like the option provided _must_ be in the array. Unless the driver supports compression mechanisms other than snappy and zlib and I'm just missing something.